src/StateManagers/CoordinatesManager.vala

```diff
@@ -76,26 +81,24 @@ public class Akira.StateManagers.CoordinatesManager : Object {
     }
 
     /**
-     * Initialize the manager coordinates with the newly created or selected item.
+     * Initialize the manager coordinates with the selected items coordinates.
+     * The coordinates change comes from a canvas action that already moved the items,
+     * therefore we set the do_update to false to prevent updating the selected
+     * items' Cairo Matrix.
      */
-    private void on_init_state_coords (Lib.Items.CanvasItem item) {
-        // Get the item X & Y coordinates.
-        double item_x = item.coordinates.x;
-        double item_y = item.coordinates.y;
+    private void on_init_state_coords () {
+        do_update = false;
 
-        // Interrupt if no value has changed.
-        if (item_x == x && item_y == y) {
-            return;
-        }
+        // Get the item X & Y coordinates.
+        x = canvas.nob_manager.selected_x;
```

Comment:
why not just request the position here? There is no real performance gain here.

Reply:
Do you mean by calling the static method?
Wouldn't that be a bit of an overkill since I only need 2 variables?
I'm definitely not against it.